examples/access_traits/example_cuda_access_traits.cpp

```diff
@@ -0,0 +1,90 @@
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <array>
+#include <iostream>
+#include <numeric>
+
+struct PointCloud
+{
+  float *d_x;
+  float *d_y;
+  float *d_z;
+  int N;
+};
+
+struct NearestToOrigin
+{
+  int k;
+};
+
+namespace ArborX
+{
+namespace Traits
+{
+template <>
+struct Access<PointCloud, PrimitivesTag>
+{
+  inline static std::size_t size(PointCloud const &cloud) { return cloud.N; }
+  KOKKOS_INLINE_FUNCTION static Point get(PointCloud const &cloud,
+                                          std::size_t i)
+  {
+    return {{cloud.d_x[i], cloud.d_y[i], cloud.d_z[i]}};
+  }
+  using memory_space = Kokkos::CudaSpace;
+};
+
+template <>
+struct Access<NearestToOrigin, PredicatesTag>
+{
+  inline static std::size_t size(NearestToOrigin const &) { return 1; }
+  KOKKOS_INLINE_FUNCTION static auto get(NearestToOrigin const &d, std::size_t)
+  {
+    return nearest(Point{{0, 0, 0}}, d.k);
+  }
+  using memory_space = Kokkos::CudaSpace;
+};
+} // namespace Traits
+} // namespace ArborX
+
+std::ostream &operator<<(std::ostream &os, ArborX::Point const &p)
+{
+  os << '(' << p[0] << ',' << p[1] << ',' << p[2] << ')';
+  return os;
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::initialize(argc, argv);
+
+  {
+    constexpr std::size_t N = 1000;
+    std::array<float, N> a;
+
+    float *d_a;
+    cudaMalloc(&d_a, sizeof(float) * N);
+
+    std::iota(std::begin(a), std::end(a), 1.0);
+
+    cudaMemcpy(d_a, a.data(), sizeof(float) * N, cudaMemcpyHostToDevice);
+
+    using device_type = Kokkos::Cuda::device_type;
+    ArborX::BVH<device_type> bvh{PointCloud{d_a, d_a, d_a, N}};
+
+    Kokkos::View<int *, device_type> indices("indices", 0);
+    Kokkos::View<int *, device_type> offsets("offsets", 0);
+    bvh.query(NearestToOrigin{5}, indices, offsets);
```

Comment:
I think it would be better to show how to do multiple queries at a time. Would not want a single query to be what user sees first.